darwin/torch/utils.py

```diff
@@ -0,0 +1,269 @@
+        split_seed: fix seed for random split creation
+
+    Output:
+        root: local path to the dataset
+        split_path: relative path to the selected train/val/test split
+    """
+    if client is None:
+        client = Client.default()
+
+    # Get data
+    dataset = None
+    local_datasets = {dataset.slug: dataset for dataset in client.list_local_datasets()}
+    if dataset_name in local_datasets:
+        if force_fetching:
+            # Remove the local copy of the dataset
+            dbpath = Path(client.project_dir, dataset_name)
+            try:
+                shutil.rmtree(dbpath)
+            except PermissionError:
+                print(
+                    "Could not remove dataset in {dbpath}. Permission denied. \
+                      Remove it manually or disable force_fetching."
+                )
+        else:
+            dataset = local_datasets[dataset_name]
+
+    if dataset is None:  # Could not find it locally or force_fetching is True
+        remote_datasets = [dataset.slug for dataset in client.list_remote_datasets()]
+        if dataset_name in remote_datasets:
+            dataset = client.get_remote_dataset(slug=dataset_name)
+            progress, _count = dataset.pull(image_status=image_status)
+            with tqdm(total=_count, desc=f"Downloading '{dataset_name}' dataset") as pbar:
+                for _ in progress():
+                    pbar.update()
+        else:
+            raise ValueError(f"Could not find dataset {dataset_name} in Darwin.")
+
+    # Find annotations and create folders
+    root = Path(client.project_dir) / dataset_name
+    annot_path = root / "annotations"
+    annot_files = [f for f in annot_path.glob("*.json")]
+    num_images = len(annot_files)
+    lists_path = root / "lists"
+    os.makedirs(lists_path, exist_ok=True)
+
+    # Extract classes from mask annotations
+    fname = lists_path / "classes_masks.txt"
+    if not fname.exists():
+        # Extract list of classes
+        classes, idx_to_classes = extract_classes(annot_files, anno_type="polygon")
+        classes_names = [k for k in classes.keys()]
+        classes_names.insert(0, "__background__")
+        with open(fname, "w") as f:
+            for c in classes_names:
+                f.write(f"{c}\n")
+
+    # Extract classes from tags
+    fname = lists_path / "classes_tags.txt"
+    if not fname.exists():
+        # Extract list of classes
+        classes, idx_to_classes = extract_classes(annot_files, anno_type="tag")
+        classes_names = [k for k in classes.keys()]
+        with open(fname, "w") as f:
+            for c in classes_names:
+                f.write(f"{c}\n")
+
+    # Create split
+    split_id = f"split_val{val_percentage}_test{test_percentage}"
+    if split_seed is not None:
+        np.random.seed(split_seed)
+        split_id += f"_seed{split_seed}"
+
+    split_path = lists_path / split_id
+    if not split_path.exists() or force_resplit:
```

Comment:
This splitting does not take into account of the classes distribution. If it is guaranteed that the classes exist (from either type of annotation) then we can use the solution proposed above in the other comment?